lib/utils/flattenNestedSelectorsForRule.mjs

```diff
@@ -7,20 +8,37 @@ import parseSelector from './parseSelector.mjs';
 /**
  * @typedef {import('postcss-selector-parser').Selector} Selector
  * @typedef {import('postcss-selector-parser').Root} Root
+ * @typedef {{selector: Selector, resolvedSelectors: Root, nested: boolean}} FlattenedSelector
  *
  * @param {import('postcss').Rule} rule
  * @param {import('stylelint').PostcssResult} result
- * @returns {Array<{selector: Selector, resolvedSelectors: Root}>}
+ * @returns {Array<FlattenedSelector>}
  */
 export default function flattenNestedSelectorsForRule(rule, result) {
 	const ownAST = parseSelector(getRuleSelector(rule), result, rule);
 
 	if (!ownAST) return [];
 
+	/** @type {Array<FlattenedSelector>} */
 	const flattenedSelectors = [];
 
 	for (const selectorAST of ownAST.nodes) {
-		const resolvedSelectors = resolveNestedSelector(selectorAST.toString(), rule);
+		const selector = selectorAST.toString();
+
+		const resolvedSelectors = resolveNestedSelector(selector, rule);
+
+		if (resolvedSelectors.length === 1 && resolvedSelectors[0] === selector) {
```

Comment:
question

When the `length === 1` shouldn't `isStandardSyntaxSelector` be called too?

Reply:
`isStandardSyntaxSelector` is called on the output of `resolveNestedSelector` to make sure that the resolved selector is still a standard selector.

For example:

```scss
.icon-#{$name} {
  .foo & {
    color: red;
  }
}
```

- `.foo &` is a standard selector
- `.foo .icon-#{$name}` is non-standard

------

We make the assumption that any input to `flattenNestedSelectorsForRule` is already checked to be a standard selector.

(This assumption is true in practice).

Since the result of calling `resolveNestedSelector` was the same as the input we can also assume that the result will be a standard selector.

Reply:
> We make the assumption that any input to flattenNestedSelectorsForRule is already checked to be a standard selector.

That's the kind of details you can add to the [`@description`](https://jsdoc.app/tags-description).
BTW you can remove the `@returns`.

Reply:
I've added a description in the doc comment but I didn't use `@description` as no other part of Stylelint uses this. I didn't want to deviate from existing practices.